src/sentry/snuba/metrics/datasource.py

```diff
@@ -678,36 +688,46 @@ def get_series(projects: Sequence[Project], metrics_query: MetricsQuery) -> dict
                     request,
                     use_cache=False,
                     referrer=f"api.metrics.{key}",
-                )["data"]
+                )
+                snuba_result_data = snuba_result["data"]
+                meta.extend(snuba_result["meta"])
 
                 snuba_limit = snuba_query.limit.limit if snuba_query.limit else None
-                if not group_limit_filters and snuba_limit and len(snuba_result) == snuba_limit:
-                    group_limit_filters = _get_group_limit_filters(metrics_query, snuba_result)
+                if (
+                    not group_limit_filters
+                    and snuba_limit
+                    and len(snuba_result_data) == snuba_limit
+                ):
+                    group_limit_filters = _get_group_limit_filters(metrics_query, snuba_result_data)
 
                     # We're now applying a filter that past queries may not have
                     # had. To avoid partial results, remove extra groups that
                     # aren't in the filter retroactively.
                     if group_limit_filters:
                         _prune_extra_groups(results, group_limit_filters)
 
-                results[entity][key] = {"data": snuba_result}
+                results[entity][key] = {"data": snuba_result_data}
 
     assert projects
     converter = SnubaResultConverter(
         projects[0].organization_id, metrics_query, fields_in_entities, intervals, results
     )
 
-    result_groups = converter.translate_results()
+    # Translate applies only on ["data"]
+    result_groups = converter.translate_result_groups()
     # It can occur, when we make queries that are not ordered, that we end up with a number of
     # groups that doesn't meet the limit of the query for each of the entities, and hence they
     # don't go through the pruning logic resulting in a total number of groups that is greater
     # than the limit, and hence we need to prune those excess groups
     if len(result_groups) > metrics_query.limit.limit:
         result_groups = result_groups[0 : metrics_query.limit.limit]
 
+    metrics_query_fields = {str(metric_field) for metric_field in metrics_query.select}
```

Comment:
```suggestion
    metrics_query_fields = set(map(str, metrics_query.select))
```

In [3]: set(map(str, a))
Out[3]: {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9'}

Reply:
I prefer the set comprehension to be honest

Reply:
sure